doc/api/deprecations.md

```diff
@@ -146,9 +146,9 @@ instead.
 <a id="DEP0012"></a>
 ### DEP0012: Domain.dispose
 
-Type: Runtime
+Type: End-of-Life
 
-[`Domain.dispose()`][] is deprecated. Recover from failed I/O actions
+[`Domain.dispose()`][] is removed. Recover from failed I/O actions
 explicitly via error event handlers set on the domain instead.
```

Comment:
Might be a good idea to expand on the description in here to explain *why* it was removed and why it is an anti-pattern. That doesn't have to be in this PR, but it would be worthwhile at some point.

Reply:
I think we had a whole document explaining why and how domains were broken, including `.dispose()`? I can’t seem to find it anywhere in our repos, though…

Reply:
heh.. there's something floating around somewhere but I cannot remember where.

Reply:
The domains postmortem is at https://nodejs.org/en/docs/guides/domain-postmortem/